flink-yarn/src/main/java/org/apache/flink/yarn/WorkerSpecContainerResourceAdapter.java

```diff
@@ -71,52 +72,65 @@
 	}
 
 	Optional<Resource> tryComputeContainerResource(final WorkerResourceSpec workerResourceSpec) {
-		return Optional.ofNullable(workerSpecToContainerResource.computeIfAbsent(
+		final InternalContainerResource internalContainerResource = workerSpecToContainerResource.computeIfAbsent(
 			Preconditions.checkNotNull(workerResourceSpec),
-			this::createAndMapContainerResource));
+			this::createAndMapContainerResource);
+		if (internalContainerResource != null) {
+			return Optional.of(internalContainerResource.toResource());
+		} else {
+			return Optional.empty();
+		}
 	}
 
 	Set<WorkerResourceSpec> getWorkerSpecs(final Resource containerResource, final MatchingStrategy matchingStrategy) {
-		return getEquivalentContainerResource(containerResource, matchingStrategy).stream()
+		final InternalContainerResource internalContainerResource = new InternalContainerResource(containerResource);
+		return getEquivalentInternalContainerResource(internalContainerResource, matchingStrategy).stream()
```

Comment:
Seems equivalent to `getEquivalentContainerResource(final Resource containerResource, final MatchingStrategy matchingStrategy)`.

Reply:
Same here. I'd like to make it explicit that `Resource` is only used in arguments and return values of methods visible from outside, and everywhere else should use `InternalContainerResource`.